csrc/scheduler/registry_utils.cpp

```diff
@@ -391,14 +391,31 @@ bool requiresForwardViewReplay(Fusion* fusion, ComputeAtMap& ca_map) {
 
 // Returns if view interferes with how we want to treat the reference, being at
 // least a 2D reduction schedule but maybe a 3D reduction schedule.
-bool reductionInterferingView(
-    Fusion* fusion,
-    const ComputeAtMap& ca_map,
-    TensorView* reduction_reference) {
+bool reductionInterferingView(Fusion* fusion, TensorView* reduction_reference) {
   // Make sure the view doesn't interfere with how we'll want to schedule
   // it. If we might want to do a 3D scheduler make sure views are disjoint
   // based on what the 3D scheduler's merges would be.
 
+  // don't need to check view ops before reduction, e.g. T1 = view(T0),
```

Comment:
Suppose there're two view ops before a reduction, e.g.:

```
t0: [I0]
// I0 -> I1, I2
t1 = reshape(t0, [I1, I2]);
// I2 -> I3, I4
t2 = reshape(t1, [I1, I3, I4]);
t3 = sum(t2, {-1);
```

And also there's another reshape using `t1`:

```
// [I1, I2] -> I5
t4 = reshape(t1, [I5])
```

The first two reshapes would be just ignored as they are before the reduction. The last reshape would be checked but since nothing would be known about `I2`, it would also be just ignored. However, the last reshape should be considered conflicting.

Am I understanding correctly?

Reply:
(1) `requiresForwardViewReplay` captures this case (two uses of I2, `I2 -> I3, I4` and `[I1, I2] -> I5`) before doing `reductionInterferingView`.
Added a test:
```
  auto tv2 = reshape(tv1, {N}, {16, 8});
  auto tv3 = reshape(tv2, {16, 8}, {16, 4, 2});
  auto tv4 = sum(tv3, {-1});
  // ID {8} was used in split, {8} --> {4, 2}
  // Now it is used again in merge, {16} * {8} --> {N}
  // requiresForwardViewReplay() captures this non-consistant uses.
  auto tv5 = reshape(tv2, {16, 8}, {N});
```
(2) slightly modified `reductionInterferingView` to check `view_ops_not_used_by_reduction` to ensure sibling view ops of reduction ops are checked. e.g. (similar to the previous case but changed `tv5 = reshape(tv2)` to `tv5 = reshape(tv3)` )
```
  auto tv2 = reshape(tv1, {N}, {16, 8});
  auto tv3 = reshape(tv2, {16, 8}, {16, 4, 2});
  auto tv4 = sum(tv3, {-1});
  // This reshape merges an Iter dim and a reduction dim
  auto tv5 = reshape(tv3, {16, 4, 2}, {N});
```

Reply:
It seems difficult to mathmatical prove that this method covers all the cases. But if some unsupported corner case happens, it fails early and loudly since `NVF_ERROR` is used to ensure all the IDs exist in `vg_to_coalesced_ids` instead of silently `continue`